internal/envtest/webhooks.go

```diff
@@ -74,11 +74,25 @@ func initWebhookInstallOptions() envtest.WebhookInstallOptions {
 	if err != nil {
 		klog.Fatalf("Failed to append controlplane controller webhook config: %v", err)
 	}
+
+	var localServingHostExternalName string
+	if strings.ToLower(os.Getenv("USE_EXISTING_CLUSTER")) == "true" {
+		if goruntime.GOOS == "darwin" {
+			// On MacOS with Docker for Mac localhost is reachable via "docker.for.mac.localhost".
+			localServingHostExternalName = "docker.for.mac.localhost"
+		}
+		if goruntime.GOOS == "linux" {
+			// docker network inspect kind [0].IPA.Config[0].Gateway
+			localServingHostExternalName = "172.18.0.1"
+		}
```

Comment:
There are two options here:
1) somehow find out which is the correct IP. Imho this is pretty fragile as we have to assume the kind cluster is using the `kind` network and then pick the gateway IP of the first IPAM config as `localServingHostExternalName`

2) an alternative would be to describe that the kind cluster has to be "prepared". This would look like this:
    * `kind create cluster`
    * `docker exec -it <container-id> "echo '172.18.0.1 linux.localhost' > /etc/hosts"` (container-id, and gateway IP has to be calculated first)
    * then we can use `linux.localhost` as our `localServingHostExternalName`

I think I would move the calculation of `localServingHostExternalName` into the hack script and then choose option 2) to "prepare" the kind cluster. The shell script would in this case print out a few commands for Linux/Mac on how to setup the test environment. This would look like:

MacOS
```bash
kind create cluster
export CAPI_WEBHOOK_HOSTNAME="docker.for.mac.localhost"
```

Linux
```bash
kind create cluster
<commands-to-prepare-kind>
export CAPI_WEBHOOK_HOSTNAME="linux.localhost"
```